external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -218,7 +224,11 @@ public void nextTuple() {
             }
 
             if (poll()) {
-                setWaitingToEmit(pollKafkaBroker());
+                try {
+                    setWaitingToEmit(pollKafkaBroker());
+                } catch (Exception e) {
+                    LOG.error("Failed to poll from kafka.", e);
```

Comment:
Right, that makes sense. We're not getting them because I think automatic assignment avoids them. I think it's fine to catch them, but you should replace the Exception with the specific exception classes you're seeing.